source-code-manager/src/main/java/org/jboss/da/scm/impl/SCMClonner.java

```diff
@@ -56,4 +67,46 @@ private ScmRepository getScmRepository(String scmUrl, ScmManager scmManager)
             throw new ScmException("Error while connecting to the repository", ex);
         }
     }
+
+    /**
+     * If we were able to shallow clone, this function will return true.
+     * false otherwise
+     *
+     * @param scmType type of scmUrl
+     * @param scmUrl link of repo to clone
+     * @param revision specific revision to clone
+     * @param cloneTo directory to clone the scmUrl
+     * @return
+     */
+    private boolean shallowClone(SCMType scmType, String scmUrl, String revision, File cloneTo) {
+
+        // we only support git for shallow cloning
+        if (!scmType.equals(SCMType.GIT)) {
+            return false;
+        }
+
+        try {
+            ProcessBuilder pb;
+            if (revision.isEmpty() || revision == null) {
```

Comment:
This could lead to NPE, either switch position of the conditions or use ``StringUtils.isBlank()`` from ``org.apache.commons.lang.StringUtils``